### PR TITLE
Update VThreadEventTest to check processor count on Windows platform

### PR DESCRIPTION
**Pull request by fengxue-IS**
Related: https://github.com/eclipse-openj9/openj9/issues/18712

### CODE REVIEW

**Comment by babsingh** [comment]
@fengxue-IS Is this ready for review?

fyi @pshipton

**Comment by pshipton** [comment]
This needs to be forward ported and delivered to head and jdk22 first, unless it's specific to jdk21 for some reason.

**Comment by fengxue-IS** [comment]
Sorry I missed the github notification.

Created the forward port for JDK22 and next

File: test/hotspot/jtreg/serviceability/jvmti/vthread/VThreadEventTest/VThreadEventTest.java [hunk 1]
```diff
@@ -136,6 +136,10 @@ private static void await(CountDownLatch dumpedLatch) {
     public static void main(String[] args) throws Exception {
         if (Runtime.getRuntime().availableProcessors() < 8) {
             log("WARNING: test expects at least 8 processors.");
+            if (System.getProperty("os.name").startsWith("Windows") && (Runtime.getRuntime().availableProcessors() < 16)) {
+                // On Windows platform, ensure sufficient processor available to avoid deadlock.
+                return;
```
Comment (pshipton) [inline test/hotspot/jtreg/serviceability/jvmti/vthread/VThreadEventTest/VThreadEventTest.java:141]: I'm thinking we should print a message that there aren't enough processors and the test isn't running, to help avoid any confusion down the line.

Comment (fengxue-IS) [reply test/hotspot/jtreg/serviceability/jvmti/vthread/VThreadEventTest/VThreadEventTest.java:142]: I've added new warning message to clarify this. Will update the other porting PRs.

**Review by pshipton — APPROVED** [review]

**Pull request merged**